src/main/java/com/denizenscript/ddiscordbot/objects/DiscordSelectionTag.java

```diff
@@ -0,0 +1,283 @@
+package com.denizenscript.ddiscordbot.objects;
+
+import com.denizenscript.denizencore.objects.Fetchable;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.objects.core.MapTag;
+import com.denizenscript.denizencore.tags.Attribute;
+import com.denizenscript.denizencore.tags.ObjectTagProcessor;
+import com.denizenscript.denizencore.tags.TagContext;
+import com.denizenscript.denizencore.tags.TagRunnable;
+import com.denizenscript.denizencore.utilities.CoreUtilities;
+import com.denizenscript.denizencore.utilities.text.StringHolder;
+import net.dv8tion.jda.api.entities.Emoji;
+import net.dv8tion.jda.api.interactions.components.selections.SelectOption;
+import net.dv8tion.jda.api.interactions.components.selections.SelectionMenu;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+
+public class DiscordSelectionTag implements ObjectTag {
+
+    // <--[ObjectType]
+    // @name DiscordSelectionTag
+    // @prefix discordselection
+    // @base ElementTag
+    // @format
+    // The identity format for Discord selection menu is a map of menu data. Do not alter raw menu data, use the with.as tag instead.
+    //
+    // @plugin dDiscordBot
+    // @description
+    // A DiscordSelectionTag is an object that represents a Discord selection menu for use with dDiscordBot.
+    //
+    // -->
+
+    @Fetchable("discordselection")
+    public static DiscordSelectionTag valueOf(String string, TagContext context) {
+        if (string.startsWith("discordselection@")) {
+            string = string.substring("discordselection@".length());
+        }
+        MapTag map = MapTag.valueOf(string, context);
+        if (map == null) {
+            return null;
+        }
+        return new DiscordSelectionTag(map);
+    }
+
+    @Override
+    public DiscordSelectionTag duplicate() {
+        return new DiscordSelectionTag(menuData.duplicate());
+    }
+
+    public static boolean matches(String arg) {
+        if (arg.startsWith("discordselection@")) {
+            return true;
+        }
+        return MapTag.matches(arg);
+    }
+
+    public DiscordSelectionTag() {
+        menuData = new MapTag();
+    }
+
+    public DiscordSelectionTag(MapTag map) {
+        menuData = map;
+    }
+
+    public static MapTag getSelectionOptions(List<SelectOption> selectOptions) {
```

Comment:
Map output / List input doesn't make sense (ref Discord discussion in actives-only)